jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/BuildImageStep.java

```diff
@@ -173,12 +174,43 @@ public DescriptorDigest getDiffId() throws LayerPropertyNotFoundException {
                 .build());
       }
       if (containerConfiguration != null) {
+        boolean shouldInheritEntrypoint = containerConfiguration.getEntrypoint() == null;
+        boolean shouldInheritProgramArguments =
+            containerConfiguration.getEntrypoint() == null
+                && (containerConfiguration.getProgramArguments() == null
+                    || containerConfiguration.getProgramArguments().isEmpty());
```

Comment:
Hmm, I think it should not inherit the base image program arguments if the program arguments configuration was set to an empty list.

Reply:
But it does not work since by default it's set to an empty list.

Reply:
Yeah, I think we should decide whether to inherit CMD or not in the same way as ENTRYPOINT.